survey/views.py

```diff
@@ -108,36 +108,41 @@ def add_checkin(request):
             else:
                 pass
     else:
-        # initialize forms with cookies
-        initial_commute = {}
-        initial_extra_commute = {}
-
-        for attr in ['name', 'email', 'home_address', 'work_address', 'employer', 'team']:
-            if attr in request.session:
-                initial_commute[attr] = request.session.get(attr)
-
-        for attr in ['share', 'comments', 'volunteer']:
-            if attr in request.session:
-                initial_extra_commute[attr] = request.session.get(attr)
-
-        if 'employer' in request.session:
-            id = request.session.get('employer')
-            try:
-                initial_commute['employer'] = Employer.objects.get(pk=id)
-            except Employer.DoesNotExist:
-                initial_commute['employer'] = ''
-        if 'team' in request.session:
-            teamid = request.session.get('team')
-            if teamid:
-                try:
-                    initial_commute['team'] = Team.objects.get(pk=teamid)
-                except Team.DoesNotExist:
-                    initial_commute['team'] = ''
+        if request.GET.get('referral'):
```

Comment:
you may want to put a comment above this line explaining that we flush out the session when we detect it is a subsequent submittal as a referral.